### PR TITLE
CI and Ruby cleanup

### PR DESCRIPTION
**Pull request by zenspider**
In preparation of my metaprogramming_fixes branch... This does a
number of things:

* Creates rake tasks
* Adds pry-byebug to allow for debugging.
* Removes gem lockfiles (doesn't affect caching!)
* Removes mysql for sqlite (33% speedup!)
* Cleans some of the ruby idioms:
  * Fix up lazy ivar access so we know it s initialized properly.
  * Don't instance eval if you don't have to.
  * Pass table_name to the relation, not the assoc extension (right?).
  * Simplify some of the metaprogramming.

### CODE REVIEW

File: lib/property_sets.rb [hunk 1]
```diff
@@ -10,16 +10,19 @@
 module PropertySets
   def self.ensure_property_set_class(association, owner_class_name)
     const_name = "#{owner_class_name}#{association.to_s.singularize.camelcase}".to_sym
+
     unless Object.const_defined?(const_name)
-      property_class = Object.const_set(const_name, Class.new(ActiveRecord::Base))
-      property_class.class_eval do
+      property_class = Class.new(ActiveRecord::Base) do
```
Comment (bquorning) [inline lib/property_sets.rb:15]: ❤️

**Review by bquorning — COMMENTED** [review]
I think you can safely remove commit ef5f55fad9218c94944a70689af090867ef078a6 from the PR.

And I am not sure aa939fa2db712bd55c89fe023676ce8bc0b080ed falls under "Ruby cleanup".

File: lib/property_sets.rb (hunk 1 above)
Comment (zenspider) [reply lib/property_sets.rb:15]: I've pushed aa939fa down to my other branch (that's on top of this one because they're too hard to separate)

**Review by pschambacher — APPROVED** [review]

**Review by bquorning — APPROVED** [review]

File: Gemfile [hunk 2]
```diff
@@ -1,9 +1,11 @@
+# TOOD: figure out how to remove redundancy between this and rails4.2.gemfile
+
```
Comment (saturnflyer) [inline Gemfile:2]: this whole file can just become `eval_gemfile 'gemfiles/rails4.2.gemfile'`

File: .gitignore [hunk 3]
```diff
@@ -1,4 +1,5 @@
 Gemfile.lock
+gemfiles/*.lock
```
Comment (saturnflyer) [inline .gitignore:2]: I think these should stay so that the next person to pick it up gets the exact versions used in these changes

**Review by saturnflyer — APPROVED** [review]

File: Gemfile (hunk 2 above)
Comment (zenspider) [reply Gemfile:2]: I tried that! it blows up because the relative path used in the inner gemspec.

File: .gitignore (hunk 3 above)
Comment (zenspider) [reply .gitignore:2]: Not sure I agree. 

It's a gem. This is not standard practice for gems... and our code SHOULD be resilient to any versions we're able to match.

Comment (zenspider) [reply .gitignore:2]: Also, we only added them last month... so this has never been a problem for us in the past afaik.

**Pull request merged**